internal/provider/kubernetes/kubernetes.go

```diff
@@ -112,6 +115,16 @@ func New(ctx context.Context, restCfg *rest.Config, svrCfg *ec.Server, resources
 		mgrOpts.Cache.SyncPeriod = ptr.To(csp)
 	}
 
+	// Limit the cache to only Envoy proxy Pods to reduce memory and sync churn.
+	// ProxyTopologyInjector is the only component that interacts with Pods.
+	if mgrOpts.Cache.ByObject == nil {
+		mgrOpts.Cache.ByObject = map[client.Object]cache.ByObject{}
+	}
+
+	mgrOpts.Cache.ByObject[&corev1.Pod{}] = cache.ByObject{
```

Comment:
does this mean is it now safe to delete hasEnvoyProxyLabels ?

Reply:
Doesn't hurt to keep it in my opinion in case there's ever a cache bug